### PR TITLE
Export Function worker internal stats via Prometheus

### PR DESCRIPTION
**Pull request by jerrypeng**
### Motivation

Collect some metrics on the performance of internal modules of the function worker.  Export the metrics via prometheus

Sample:

pulsar_function_worker_rebalance_execution_time_total_ms{cluster="standalone",quantile="0.5",} 4.845677
pulsar_function_worker_rebalance_execution_time_total_ms{cluster="standalone",quantile="0.9",} 5.827341
pulsar_function_worker_rebalance_execution_time_total_ms{cluster="standalone",quantile="0.99",} 5.827341
pulsar_function_worker_rebalance_execution_time_total_ms{cluster="standalone",quantile="0.999",} 5.827341
pulsar_function_worker_rebalance_execution_time_total_ms_count{cluster="standalone",} 20.0
pulsar_function_worker_rebalance_execution_time_total_ms_sum{cluster="standalone",} 121.93617499999999
pulsar_function_worker_start_up_time_ms_count{cluster="standalone",} 1.0
pulsar_function_worker_start_up_time_ms_sum{cluster="standalone",} 1470.998066
pulsar_function_worker_rebalance_strategy_execution_time_ms{cluster="standalone",quantile="0.5",} 0.129056
pulsar_function_worker_rebalance_strategy_execution_time_ms{cluster="standalone",quantile="0.9",} 0.267288
pulsar_function_worker_rebalance_strategy_execution_time_ms{cluster="standalone",quantile="0.99",} 0.267288
pulsar_function_worker_rebalance_strategy_execution_time_ms{cluster="standalone",quantile="0.999",} 0.267288
pulsar_function_worker_rebalance_strategy_execution_time_ms_count{cluster="standalone",} 20.0
pulsar_function_worker_rebalance_strategy_execution_time_ms_sum{cluster="standalone",} 4.016873
pulsar_function_worker_stop_instance_process_time_ms{cluster="standalone",quantile="0.5",} 4.096286
pulsar_function_worker_stop_instance_process_time_ms{cluster="standalone",quantile="0.9",} 5.093977
pulsar_function_worker_stop_instance_process_time_ms{cluster="standalone",quantile="0.99",} 5.093977
pulsar_function_worker_stop_instance_process_time_ms{cluster="standalone",quantile="0.999",} 5.093977
pulsar_function_worker_stop_instance_process_time_ms_count{cluster="standalone",} 8.0
pulsar_function_worker_stop_instance_process_time_ms_sum{cluster="standalone",} 42.296554
pulsar_function_worker_start_instance_process_time_ms{cluster="standalone",quantile="0.5",} 111.921888
pulsar_function_worker_start_instance_process_time_ms{cluster="standalone",quantile="0.9",} 141.08962
pulsar_function_worker_start_instance_process_time_ms{cluster="standalone",quantile="0.99",} 141.08962
pulsar_function_worker_start_instance_process_time_ms{cluster="standalone",quantile="0.999",} 141.08962
pulsar_function_worker_start_instance_process_time_ms_count{cluster="standalone",} 9.0
pulsar_function_worker_start_instance_process_time_ms_sum{cluster="standalone",} 1123.083107
pulsar_function_worker_instance_count{cluster="standalone",} 1.0
pulsar_function_worker_schedule_strategy_execution_time_ms{cluster="standalone",quantile="0.5",} 0.006223
pulsar_function_worker_schedule_strategy_execution_time_ms{cluster="standalone",quantile="0.9",} 0.03648
pulsar_function_worker_schedule_strategy_execution_time_ms{cluster="standalone",quantile="0.99",} 0.03648
pulsar_function_worker_schedule_strategy_execution_time_ms{cluster="standalone",quantile="0.999",} 0.03648
pulsar_function_worker_schedule_strategy_execution_time_ms_count{cluster="standalone",} 4.0
pulsar_function_worker_schedule_strategy_execution_time_ms_sum{cluster="standalone",} 0.10072700000000001
pulsar_function_worker_schedule_execution_time_total_ms{cluster="standalone",quantile="0.5",} 152.412083
pulsar_function_worker_schedule_execution_time_total_ms{cluster="standalone",quantile="0.9",} 152.412083
pulsar_function_worker_schedule_execution_time_total_ms{cluster="standalone",quantile="0.99",} 152.412083
pulsar_function_worker_schedule_execution_time_total_ms{cluster="standalone",quantile="0.999",} 152.412083
pulsar_function_worker_schedule_execution_time_total_ms_count{cluster="standalone",} 4.0
pulsar_function_worker_schedule_execution_time_total_ms_sum{cluster="standalone",} 1051.835816

### CODE REVIEW

File: pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/WorkerStatsManager.java [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+  private static final String PULSAR_FUNCTION_WORKER_METRICS_PREFIX = "pulsar_function_worker_";
+  private static final String START_UP_TIME = "start_up_time_ms";
+  private static final String INSTANCE_COUNT = "instance_count";
+  private static final String SCHEDULE_TOTAL_EXEC_TIME = "schedule_execution_time_total_ms";
+  private static final String SCHEDULE_STRATEGY_EXEC_TIME = "schedule_strategy_execution_time_ms";
+  private static final String REBALANCE_TOTAL_EXEC_TIME = "rebalance_execution_time_total_ms";
+  private static final String REBALANCE_STRATEGY_EXEC_TIME = "rebalance_strategy_execution_time_ms";
+  private static final String STOPPING_INSTANCE_PROCESS_TIME = "stop_instance_process_time_ms";
+  private static final String UPDATING_INSTANCE_PROCESS_TIME = "update_instance_process_time_ms";
+  private static final String STARTING_INSTANCE_PROCESS_TIME = "start_instance_process_time_ms";
```
Comment (srkukarni) [inline pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/WorkerStatsManager.java:42]: Do we also want to measure number of function instances started/stopped/terminated.
What about differentiating between sources/sinks/functions?

Comment (jerrypeng) [reply pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/WorkerStatsManager.java:42]: > Do we also want to measure number of function instances started/stopped/terminated.
A count is already include e.g. 

pulsar_function_worker_stop_instance_process_time_ms_count 10

> What about differentiating between sources/sinks/functions?

Let me think more on that.  The thing is this is more of a backend metric (Assignment processing workflow).  We don't really differentiate between sources/sinks/functions at that level.

**Pull request merged**